src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -433,7 +433,10 @@ public final void run() {
                         // in an immediately ready selection key, hence we use the non-blocking form
                         selected = selector.selectNow();
                     } else {
-                        selected = selector.select(); // WINDOWS!!! Waiting the whole timeout anyway if timeout specified?!?
```

Comment:
this was by design.

if you wait for 1 second then on windows if something happens at t=0 the selector does not return (it should return immedatley something is available),  it will return all the keys that are selecteded at T=1s

Reply:
fixed